geomstats/geometry/positive_reals.py

```diff
@@ -0,0 +1,315 @@
+
+        Returns
+        -------
+        samples : array-like, shape=[..., 1]
+            Points sampled in the positive reals.
+        """
+        size = (n_samples, 1) if n_samples != 1 else (1,)
+        samples = bound * gs.random.rand(*size)
+        samples = gs.maximum(samples, atol)
+        return samples
+
+
+class PositiveRealsMetric(RiemannianMetric):
+    """Class for the positive reals metric.
+
+    It is a particular case in dimension 1
+    of the SPD affine-invariant metric
+    with a power affine coefficient equal to one.
+
+    Parameters
+    ----------
+    scale : float
+        Scale of the positive reals metric.
+        Optional, default: 1.
+    """
+
+    def __init__(self, scale=1.0):
+        self.scale = scale
+        super().__init__(
+            dim=1,
+            shape=(1,),
+            signature=(1, 0),
+        )
+
+    def inner_product_matrix(self, base_point):
+        """Compute the inner product matrix at base point.
+
+        Parameters
+        ----------
+        base_point : array-like, shape=[..., 1]
+            Base point.
+
+        Returns
+        -------
+        inner_prod_mat : array-like, shape=[..., 1]
+            Inner product matrix.
+        """
+        inner_prod_mat = 1 / base_point**2
+        inner_prod_mat *= self.scale**2
+        return inner_prod_mat
+
+    def inner_product(self, tangent_vec_a, tangent_vec_b, base_point):
+        """Compute the positive reals inner-product.
+
+        Compute the inner-product of tangent_vec_a and tangent_vec_b
+        at point base_point using the positive reals Riemannian metric.
+
+        Parameters
+        ----------
+        tangent_vec_a : array-like, shape=[..., 1]
+            Tangent vector at base point.
+        tangent_vec_b : array-like, shape=[..., 1]
+            Tangent vector at base point.
+        base_point : array-like, shape=[..., 1]
+            Base point.
+
+        Returns
+        -------
+        inner_product : array-like, shape=[...,]
+            Inner-product.
+        """
+        inner_product_matrix = self.inner_product_matrix(base_point=base_point)
+        inner_product = tangent_vec_a * inner_product_matrix * tangent_vec_b
+        inner_product = gs.reshape(inner_product, (-1,))
+        return inner_product
+
+    def squared_norm(self, vector, base_point):
+        """Compute the square of the norm of a vector.
+
+        Squared norm of a vector associated to the inner product
+        at the tangent space at a base point.
+
+        Parameters
+        ----------
+        vector : array-like, shape=[..., 1]
+            Vector.
+        base_point : array-like, shape=[..., 1]
+            Base point.
+            Optional, default: None.
+
+        Returns
+        -------
+        sq_norm : array-like, shape=[...,]
+            Squared norm.
+        """
+        sq_norm = self.inner_product(vector, vector, base_point=base_point)
+        return gs.real(sq_norm)
```

Comment:
shouldn't the result be real already?

Reply:
Once `PositiveReals` and `ComplexPoincareDisk` are both merged, I will create a product manifold named `ProductPositiveRealsAndComplexPoincareDisks`. 
In this product manifold, the default dtype is complex.
Therefore, it is very convenient if `PositiveReals` can accept both reals a complex data types.